projects/demo-cypress/src/tests/lazy-loading.cy.ts

```diff
@@ -0,0 +1,75 @@
+
+describe('Tabs', () => {
+    let component: TestComponent;
+    let fixture: ComponentFixture<TestComponent>;
+
+    @Component({
+        template: `
+            <tui-root>
+                <img
+                    alt="picsum"
+                    loading="lazy"
+                    style="--tui-clear-hover: rgba(0, 0, 0, 0.16); width: 100px; height: 100px; display: block"
+                    [attr.src]="src"
+                />
+            </tui-root>
+        `,
+    })
+    class TestComponent {
+        @Input()
+        src = '';
+    }
+
+    beforeEach(() =>
+        cy
+            .mount(TestComponent, {
+                imports: [TuiRootModule, TuiLazyLoadingModule],
+                componentProperties: {
+                    src: '',
+                },
+            })
+            .then(wrapper => {
+                component = wrapper.component;
+                fixture = wrapper.fixture;
+            }),
+    );
+
+    it('image has background color', () => {
+        cy.get('img').then(query => {
+            const img = query.get(0);
+
+            expect(img.style.background).to.equal('var(--tui-clear-hover)');
+            expect(img.style.getPropertyValue('--tui-clear-hover').trim()).to.equal(
+                'rgba(0, 0, 0, 0.16)',
+            );
+        });
+    });
+
+    it('loading animation is shown', () => {
+        cy.get('img').then(query => {
+            const img = query.get(0);
+
+            expect(img.style.animationName).to.equal('tuiSkeletonVibe');
+        });
+    });
+
+    it('loading animation is cancelled after image load', () => {
+        cy.get('img').then(query => {
+            const img = query.get(0);
+
+            expect(img.style.animationName).to.equal('tuiSkeletonVibe');
+
+            component.src =
+                'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/4QDeRXhpZgAASUkqAAgAAAAGABIBAwABAAAAAQAAABoBBQABAAAAVgAAABsBBQABAAAAXgAAACgBAwABAAAAAgAAABMCAwABAAAAAQAAAGmHBAABAAAAZgAAAAAAAABIAAAAAQAAAEgAAAABAAAABwAAkAcABAAAADAyMTABkQcABAAAAAECAwCGkgcAFgAAAMAAAAAAoAcABAAAADAxMDABoAMAAQAAAP//AAACoAQAAQAAAAEAAAADoAQAAQAAAAEAAAAAAAAAQVNDSUkAAABQaWNzdW0gSUQ6IDY3Nf/bAEMACAYGBwYFCAcHBwkJCAoMFA0MCwsMGRITDxQdGh8eHRocHCAkLicgIiwjHBwoNyksMDE0NDQfJzk9ODI8LjM0Mv/bAEMBCQkJDAsMGA0NGDIhHCEyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMv/CABEIAAEAAQMBIgACEQEDEQH/xAAVAAEBAAAAAAAAAAAAAAAAAAAABf/EABUBAQEAAAAAAAAAAAAAAAAAAAAB/9oADAMBAAIQAxAAAAGOE//EABQQAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQEAAQUCf//EABQRAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQMBAT8Bf//EABQRAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQIBAT8Bf//EABQQAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQEABj8Cf//EABQQAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQEAAT8hf//aAAwDAQACAAMAAAAQC//EABQRAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQMBAT8Qf//EABQRAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQIBAT8Qf//EABQQAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQEAAT8Qf//Z';
+            fixture.detectChanges();
+
+            cy.wait(100).then(() => {
+                expect(img.style.animationName).to.equal('');
```

Comment:
Could we use the following auto-retrying assertion instead ?

```ts
cy.get('img').should('have.css', 'animation-name', '')
```

Reply:
fixed